packages/recommend-core/src/personalization/isPersonalized.ts

```diff
@@ -0,0 +1,8 @@
+import { PersonalizationProps } from './types';
+
+export const isPersonalized = (object: any): object is PersonalizationProps => {
+  return (
+    (object.region === 'eu' || object.region === 'us') &&
+    typeof object.userToken === 'string'
+  );
```

Comment:
The user intent to enable personalization is to provide values for `region` and `userToken`, but not necessarily correct values. Otherwise this mixes the feature activation with feature validation. So we should rather check whether these values are defined.

Reply:
IMO this function is a type guard, so it should do its very best to guarantee the parameter object is of type `PersonalizationProps` 

If we check just for existence `region` and `userToken` it could provide false type guarantees to code that depend on it.

Reply:
It's used differently at 2 places though:

https://github.com/algolia/recommend/blob/b717770c11668d08fab1453de4c99edfbc12d07a/packages/recommend-js/src/experimental-personalization/frequentlyBoughtTogether.tsx#L41-L43

And in another function, it's used differently:

https://github.com/algolia/recommend/blob/b717770c11668d08fab1453de4c99edfbc12d07a/packages/recommend-react/src/experimental-personalization/Recommend.tsx#L91-L93

If the developer makes a mistake in the region, the code path for personalization will not be run. It kinda fails silently. Same would happen if a new region is introduced; the check in the library prevents it from running. We shouldn't have too many runtime type checks for search parameters, we should let the API throw if needed.

We could perhaps do:

```tsx
// usage
const { userToken, region } = validatePersonalizationProps(props)
```

```tsx
// implementation
export function isPersonalizationEnabled(props) {
  return (
    typeof props.region !== 'undefined' && typeof props.userToken !== 'undefined'
  );
}

export function getPersonalizationProps(props) {
  if (isPersonalizationEnabled(props)) {
    return {
      region: props.region,
      userToken: props.userToken,
    };
  }

  return {};
}
```

Reply:
I like `getPersonalizationProps` removes redundant checks !